wyvern/exceptions.py

```diff
@@ -19,6 +21,16 @@ def __init__(
         except Exception:
             # at least get the core message out if something happened
             self._error_string = self.message
+        wyvern_request = request_context.current()
+        request_id = None
+        if wyvern_request and wyvern_request.request_id:
+            request_id = wyvern_request.request_id
+        elif "request_id" in kwargs:
+            request_id = kwargs["request_id"]
+
+        self.request_id = request_id
+        if self.request_id:
+            self._error_string = f"{self._error_string} (Request ID: {self.request_id})"
```

Comment:
nit: How about having the request id first and defining a log processing rule in datadog to move it into a tag?

```suggestion
            self._error_string = f"[wv.request_id={self.request_id}]{self._error_string}"
```

Reply:
@ykeremy 
good idea i'll update.

eventually, I would like to have structured logging, where the log is actually a JSON object, so that `request_id` is one of the key in default key in our logs. but this is good for now

this is also going to be in the response payload of a bad pipeline request:
`{"error": error_string}`